Add x11 service for unclutter-xfixes

erikryb opened the pull request:
###### Motivation for this change

This is a service similar to the unclutter service, but for the new unclutter-xfixes. There is some [effort](https://github.com/Airblader/unclutter-xfixes/issues/14) to make unclutter-xfixes compatible with unclutter, but for now it is probably best to keep them as two different modules.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

bjornfor commented on nixos/modules/services/x11/unclutter-xfixes.nix:
```diff
@@ -0,0 +1,58 @@
+      default = 1;
+    };
+
+    threeshold = mkOption {
```
Typo (one 'e' too many).

bjornfor commented on nixos/modules/services/x11/unclutter-xfixes.nix:
```diff
@@ -0,0 +1,58 @@
+    };
+
+    threeshold = mkOption {
+      description = "Minimum number of pixels considered cursor movement";
```
Descriptions should end with a dot (full sentences).

groxxda commented on nixos/modules/services/x11/unclutter-xfixes.nix:
```diff
@@ -0,0 +1,58 @@
+  config = mkIf cfg.enable {
+    systemd.user.services.unclutter-xfixes = {
+      description = "unclutter-xfixes";
+      wantedBy = [ "default.target" ];
```
Minor nitpick: Imo this should be `graphical.target`.

erikryb replied:
Yeah, that makes sense. Better to be explicit.

groxxda commented:
We also might want some ordering `After=display-manager.service` and possibly `Requisite=display-manager.service` (?).

erikryb commented:
Why "Requisite" instead of "Requires"?

groxxda commented:
Well, in my opinion if `display-manager` is not pulled in on it's own, `unclutter` should respect that and just refuse to start.. `PartOf` or `BindsTo` could also be valid choices, that's why I added a :question: :wink:

erikryb commented:
I see. It seems that `BindsTo` is a stronger version of `Requires`, so 
that should also not be used for the same reason. Am I correct in that 
`PartOf` is a weaker version of `Requisite`, that is, does `Requisite` 
also restart or stop `unclutter` when `display-manager` restarts or 
stops? If so, I believe `Requisite` is the best option, as you first 
stated.

On Thu, Sep 08, 2016 at 05:34:35AM -0700, Alexander Ried wrote:
>Well, in my opinion if `display-manager` is not pulled in on it's own, `unclutter` should respect that and just refuse to start.. `PartOf` or `BindsTo` could also be valid choices, that's why I added a :question: :wink:
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/18398#issuecomment-245582800

groxxda commented:
I did not try the exact behavior, but `Requisite` sounded appropriate to me. I expect `clutter` to fail anyway when `display-manager` stops / fails / restarts, so it's not really that important whether it's propagated by systemd. But feel free to try and report what happens :+1: 

Apart from this, if you look at the `redshift` service that's quite similar, it will get merged even if it's not perfect.. :disappointed:

erikryb committed:
Add x11 service for unclutter-xfixes

romildo committed:
lumina: init at 1.1.0-p1

FlorentBecker committed:
pdfjam: remove obsolete version

pdfjam is included in texlive
fixes #19826

8573 committed:
agrep: init at 3.41.5

In category `tools`, subcategory `text`, add a package definition for
the program [`agrep`] [1] — "Approximate `grep` for fast fuzzy string
searching".

I have tested this patch per nixpkgs manual section 11.1 ("Making
patches").

[1]: <https://www.tgries.de/agrep/>

joachifm committed:
lttng-modules: 2.8.0 -> 2.8.3

Fixes build against linux >=4.8

Full changelog at
https://git.lttng.org/?p=lttng-tools.git;a=blob_plain;f=ChangeLog;hb=13dc409a5ea439b96b805c3c71886a3fcfad18e8

Tested with nix-build -A linuxPackages.lttng-modules -A linuxPackages_latest.lttng-modules

joachifm committed:
Merge pull request #20077 from FlorentBecker/remove_pdfjam

pdfjam: remove obsolete version

joachifm committed:
Merge pull request #20178 from 8573/8573/pkg/add/agrep/1

agrep: init at 3.41.5

joachifm committed:
Merge pull request #20063 from romildo/new.lumina

lumina: init at 1.1.0-p1

joachifm committed:
Merge pull request #20216 from joachifm/lttng-modules

lttng-modules: 2.8.0 -> 2.8.3

bjornfor commented:
Why did you close?

erikryb commented:
It was a mistake. I rebased my master branch and the pull request got closed. I guess that's why I should have used a dedicated branch for the pull request. How do I reopen it? Or should I just start a new pull request?

joachifm commented:
@erikryb I rebased & merged this at 2f0cc0d3f055d64ca3f4c8664c12e8f02a833d0d. Thank you

erikryb commented:
Thanks!
